### PR TITLE
BC::Tuple#argument accepts an optional block to define the attribute class dynamically

### PR DESCRIPTION
**Pull request by DmitryTsepelev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -344,6 +344,16 @@ module Registration
     # defines a reader and applies validation on `.match` call
     attribute :login, Email.or_a(Phone)
     attribute :password, Ascii
+    # defines an attribute using the anonimous type class
```
Comment (sclinede) [inline README.md]: anonymous

File: lib/blood_contracts/core/tuple.rb [hunk 2]
```diff
@@ -65,6 +71,18 @@ def inherited(new_klass)
...
+      # Generate an anonimous type from a block
+      #
+      # @param Proc block which will be evaluated in a context
+      # of our anonimous type
+      #
+      # @return Class
+      #
+      def type_from_block(&block)
+        raise ArgumentError unless block_given?
+        Class.new(Refined) { class_eval(&block) }
```
Comment (sclinede) [inline lib/blood_contracts/core/tuple.rb]: Let's give it a name, we have name of the attribute anyway. WDYT?
`const_set("InlineType_{name}", Class.new(Refined) { class_eval(&block) })`

**Review by sclinede — COMMENTED** [review]
Nice catch! Couple changes to go)

**Comment by sclinede** [comment]
And also `bundle exec rubocop -a` :) just to be lint perfect

**Comment by DmitryTsepelev** [comment]
Done

**Comment by sclinede** [comment]
Perfect! 🎉

**Pull request merged**